js/script.js

```diff
@@ -12,29 +21,51 @@ const addButtonHandler = () => {
   const taskTitle = sanitizeInput($taskInput.value);
   if (taskTitle) {
     createTask(taskTitle);
-    showToastMessage("Task added successfully!");
+    showToastMessage(successMessage);
     $addButton.disabled = true;
   } else {
-    showToastMessage("Please provide a valid title!");
+    showToastMessage(invalidMessage);
+  }
+};
+
+const handleSearchTasks = () => {
+  // Sanitize and retrieve the input value
+  const searchTitle = sanitizeInput($searchInput.value.trim());
+  // Clear the input field
+  $searchInput.value = "";
+  // Use early return to handle empty search scenario
+  if (!searchTitle) {
+    renderTasks(tasks);
+    return;
+  }
+  // Filter tasks based on the sanitized input
+  const filteredTasks = tasks.filter((task) =>
+    task.title.toLowerCase().includes(searchTitle.toLowerCase())
+  );
+  // Render based on the filtering result
+  if (filteredTasks.length > 0) {
+    renderTasks(filteredTasks);
+  } else {
```

Comment:
Although adding comments in code might help now and then, adding comments too much/unnecessarily will produce what is known as "code smell". This is a general anti pattern that is often not advised.

Only add comment where it is absolutely necessary. Otherwise we should write self documented code. This is one of the reasons why we strive to write simpler, modular code and avoid complex oneliners which might impress myself(the developer) and help me have a good nights sleep but will make me question the same code 5 months later.
:p